File: cinder/Entrainment/src/ViewGradientMap.cpp
//
//  ViewGradientMap.cpp
//  Entrainment03
//
//  Created by Yi-Wen Lin on 2015/7/27.
//
//

#include "ViewGradientMap.h"
#include "Utils.h"
#include "MeshUtils.h"

using namespace ci;
using namespace ci::app;
using namespace std;
using namespace bongiovi::utils;

ViewGradientMap::ViewGradientMap() : ViewCopy("shaders/copy.vert", "shaders/gradientMap.frag") {
    _textureGradient = Utils::createTexture("common/gradientMap.png");
//    _init();
}

ViewGradientMap::ViewGradientMap(string vsPath, string fsPath) : ViewCopy(vsPath, fsPath) {
//    _init();
}

void ViewGradientMap::render(gl::TextureRef texture) {
    render(*texture);
}

void ViewGradientMap::render(gl::Texture texture) {
    shader->bind();
    shader->uniform("texture", 0);
    shader->uniform("textureMap", 1);
    texture.bind(0);
    _textureGradient->bind(1);
    gl::draw(mesh);
    texture.unbind();
    _textureGradient->unbind();
    shader->unbind();
}

File: webgl/src/js/ViewGradientMap.js
// ViewGradientMap.js
var GL = bongiovi.GL;
var gl;
var glslify = require("glslify");

function ViewGradientMap() {
	bongiovi.View.call(this, null, glslify("../shaders/gradientMap.frag"));
}

var p = ViewGradientMap.prototype = new bongiovi.View();
p.constructor = ViewGradientMap;


p._init = function() {
	gl = GL.gl;
	this.textureMap = new bongiovi.GLTexture(images.gradientMap);
	this.mesh = bongiovi.MeshUtils.createPlane(2, 2, 1);
};

p.render = function(texture) {
	this.shader.bind();
	this.shader.uniform("texture", "uniform1i", 0);
	this.shader.uniform("textureMap", "uniform1i", 1);
	texture.bind(0);
	this.textureMap.bind(1);
	GL.draw(this.mesh);
};

module.exports = ViewGradientMap;